test/NITTA/Intermediate/Simulation/Tests.hs

```diff
@@ -36,7 +38,22 @@ simulationTests =
                 l1 = loop 0 "b2" ["a1"]
                 l2 = loop 1 "c" ["b1", "b2"]
                 a = add "a1" "b1" ["c"]
-            mapM_ (([l1, l2, a] @=?) . f) $ permutations [l1, l2, a]
+            mapM_ (([l1, l2, a] @=?) . f) [[l1, l2, a], [l1, a, l2], [a, l1, l2]]
+        , testGroup
+            "estimate waves"
+            [ testCase "function sequence" $ do
+                let f = estimateWaves :: [F String Int] -> [String] -> M.Map String Int
+                    a1 = add "a" "b" ["c"]
+                    a2 = add "c" "b" ["d"]
+                    a3 = add "d" "a" ["e"]
+                mapM_ ((M.fromList [("c", 0), ("d", 1), ("e", 2)] @=?) . (`f` ["a", "b"])) $ permutations [a1, a2, a3]
```

Comment:
point-free is a bad solution some times…. Very hard to read.

Reply:
I rewrote this section of tests, please check. Renamed f to action and made code less point-free

Reply:
Our hlint did not appreciate my attempt to make the code more readable, so I returned it as it was

Reply:
hlint is just a recommendation. See in https://github.com/ryukzak/nitta/blob/master/.hlint.yaml how we can suppress what we don’t like.

Reply:
Got it. I've added ignore option to `.hliny.yaml` and rewrote this code.